lib/introspector.js

```diff
@@ -88,48 +101,36 @@ const splitComments = lines => {
 };
 
 const separateCommentTypes = comment => {
-  const regex = new RegExp(/<[\w.]+(\[])?>/);
-  const regexArr = new RegExp(/\[ ?<[\w.]+(\[])?>(, ?<[\w.]+(\[])?>)* ?]/);
-
-  let typeLine = '';
+  const types = [];
   while (true) {
-    if (
-      comment.startsWith('<') ||
-      (comment.startsWith(' | ') && comment.indexOf('<') === 3)
-    ) {
-      const start = comment.indexOf('<');
-      const end = comment.indexOf('>') + 1;
-      if (!regex.test(comment.slice(start, end))) break;
-      typeLine += comment.slice(0, end);
-      comment = comment.slice(end);
-    } else if (
-      comment.startsWith('[') ||
-      (comment.startsWith(' | ') && comment.indexOf('[') === 3)
-    ) {
-      const start = comment.indexOf('[');
-      const end = comment.lastIndexOf(']') + 1;
-      if (!regexArr.test(comment.slice(start, end))) break;
-      typeLine += comment.slice(0, end);
-      comment = comment.slice(end);
+    if (comment.startsWith(' | ')) comment = comment.slice(3);
+    else if (comment.startsWith('|')) comment = comment.slice(1);
+    const start = 0;
+    let end;
+    if (comment[0] === '<') {
```

Comment:
Yeah, but there is nothing to do anyway if that validation fails (silently skipping is even worse in this case as it's impossible to understand the issue with the type), so it's up to the user to make them correct, no?

Reply:
Well, I would suggest throwing an error like a generator does or at the very least log the error.

Reply:
So how do you suggest to verify these since they can now contain multiple `<`,`>`, `[`,`]`,`{`,`}` and much more complex types?
I don't think it will be easy to do with regexp anymore.

Reply:
Yeah, we'll need to recursively parse types since they can contain other types. I'm ok with postponing the validation of the inner types. But we need to at least bring back validation of simple types like `<string>` or `[<string>, <number[]>]` as well as `<Set<{ number, string }>>`, but we won't validate `{ number, string }` part. Otherwise, it would be the same as just removing validation.

Reply:
But there is no way to distinguish the simple case of `<string>` vs a complex type. The regex will just fail and we won't be able to distinguish these. Or am I missing something?